src/msgpack_unpacker.erl

```diff
@@ -233,7 +233,8 @@ unpack_str_or_raw(V, ?OPTION{spec=new,
     {case UnpackStr of
          as_binary when ValidateString -> unpack_str(V), maybe_bin(V, Opt);
          as_binary -> maybe_bin(V, Opt);
-         as_list -> unpack_str(V)
+         as_list -> unpack_str(V);
+         with_tag -> {string, unpack_str(V)}
```

Comment:
Why would we want tagged list/string representation instead of, say, tagged binary?